client/components/domains/use-my-domain/utilities/get-option-info.js

```diff
@@ -249,9 +249,5 @@ export function getOptionInfo( {
 
 	connectContent.primary = ! transferContent?.primary;
 
-	if ( transferContent?.primary ) {
-		return [ transferContent, connectContent ];
-	}
-
-	return [ { ...connectContent, primary: true }, transferContent ];
+	return [ transferContent, connectContent ];
```

Comment:
Is this to prevent the sections from reordering? Transfer appears to me be worthy of first place, but I'll defer to whatever advice you've sought.